swarm.go

```diff
@@ -115,25 +115,21 @@ func NewSwarm(config *api.Config, mockStore *mock.NodeStore) (self *Swarm, err e
 	if config.SwapEnabled {
 		// for now, Swap can only be enabled in a whitelisted network
 		if self.config.NetworkID != swap.AllowedNetworkID {
-			return nil, fmt.Errorf("swap can only be enabled under Network ID %d, found Network ID %d instead", swap.AllowedNetworkID, self.config.NetworkID)
+			return nil, fmt.Errorf("swap can only be enabled under BZZ Network ID %d, found Network ID %d instead", swap.AllowedNetworkID, self.config.NetworkID)
 		}
-		// if Swap is enabled, we MUST have a contract API
-		if self.config.SwapBackendURL == "" {
-			return nil, errors.New("swap enabled but no contract address given; fatal error condition, aborting")
-		}
-		log.Info("connecting to SWAP API", "url", self.config.SwapBackendURL)
-		self.backend, err = ethclient.Dial(self.config.SwapBackendURL)
-		if err != nil {
-			return nil, fmt.Errorf("error connecting to SWAP API %s: %s", self.config.SwapBackendURL, err)
-		}
-
-		// initialize the balances store
-		swapStore, err := state.NewDBStore(filepath.Join(config.Path, "swap.db"))
+		//TODO: is this needed here?
+		self.backend, _ = ethclient.Dial(self.config.SwapBackendURL)
```

Comment:
@holisticode , I left this in here, but it seems to be redundant, as the backend is already stored in SWAP. Shall I remove this and adjust tests, or do we leave it in for now?

Reply:
No, I think this can (and should) be removed now, as you are already dialing in inside `New`

Reply:
The comment was about _self_.backend. Note, this is not _swap_.backend. I asked you whether self.backend is needed (a backend variable on Swarm), but I don't think it is the case and will remove it.

Reply:
Yes I know, but you are dialing in from `swap.New` now, which is the same backend, so `swam.backend` can and should indeed  be removed